reference/conanfile/methods/requirements.rst

```diff
@@ -120,4 +120,22 @@ direct
 ~~~~~~
 
 If the dependency is a direct one, that is, it has explicitly been declared by the current
-recipe, or if it is a transitive one.
\ No newline at end of file
+recipe, or if it is a transitive one.
+
+
+.. _reference_conanfile_package_type_trait_inferring:
+
+package_type trait inferring
+============================
+
+Some traits are automatically inferred based on the value of the ``package:type`` if not explicitly set by the recipe.
+
+ * ``application``: ``headers=False``, ``libs=False``, ``run=True``
+ * ``shared-library``: ``run=True``
+ * ``static-library``: ``run=False``
```

Comment:
shared and static libraries will also have ``headers=True`` and ``libs=True`` by default. 
Maybe the missing piece here is also the default traits of ``requires``, ``tool_requires`` and ``test_requires``?